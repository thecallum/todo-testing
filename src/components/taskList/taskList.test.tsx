import { render } from "@testing-library/react";
import TaskList from "./taskList";
import { Task } from '../../interfaces/interfaces'

describe("TaskList", () => {
  it("Renders component when many tasks", () => {
    // Arrange
    const tasks = [
      { value: "task description one", completed: false },
      { value: "task description two", completed: true },
    ];
    const toggleCheckbox = () => {};
    const deleteTask = () => {};

    // Act
    const { asFragment } = render(
      <TaskList
        tasks={tasks}
        toggleCheckbox={toggleCheckbox}
        deleteTask={deleteTask}
      />
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
  });

  it("Renders component when no tasks", () => {
    // Arrange
    const tasks: Task[] = [];
    const toggleCheckbox = () => {};
    const deleteTask = () => {};

    // Act
    const { asFragment } = render(
      <TaskList
        tasks={tasks}
        toggleCheckbox={toggleCheckbox}
        deleteTask={deleteTask}
      />
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
  });
});

