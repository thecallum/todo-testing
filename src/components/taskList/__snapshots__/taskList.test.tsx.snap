// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList Renders component when many tasks 1`] = `
<DocumentFragment>
  <ul
    class="taskList"
  >
    <li
      class="taskList-item"
      data-cy="task"
    >
      <div
        class="taskList-item-left"
      >
        <input
          class="taskList-item-input"
          data-cy="task-checkbox"
          type="checkbox"
        />
        <span
          class="taskList-item-value "
          data-cy="task-value"
        >
          task description one
        </span>
      </div>
      <button
        aria-label="Delete task"
        class="taskList-item-button"
        data-cy="delete-task-button"
      />
    </li>
    <li
      class="taskList-item"
      data-cy="task"
    >
      <div
        class="taskList-item-left"
      >
        <input
          checked=""
          class="taskList-item-input"
          data-cy="task-checkbox"
          type="checkbox"
        />
        <span
          class="taskList-item-value completed"
          data-cy="task-value"
        >
          task description two
        </span>
      </div>
      <button
        aria-label="Delete task"
        class="taskList-item-button"
        data-cy="delete-task-button"
      />
    </li>
  </ul>
</DocumentFragment>
`;

exports[`TaskList Renders component when no tasks 1`] = `
<DocumentFragment>
  <div
    class="taskList-empty"
    data-cy="no-tasks-message"
  >
    You have no tasks
  </div>
</DocumentFragment>
`;
